pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/TimeBuckets.java

```diff
@@ -67,13 +67,11 @@ public int resolveIndex(long timeValue) {
     if (_timeBuckets.length == 0) {
       return -1;
     }
-    if (timeValue < _timeBuckets[0]) {
+    if (timeValue <= getTimeRangeStartExclusive() || timeValue > getTimeRangeEndInclusive()) {
       return -1;
     }
-    if (timeValue >= _timeBuckets[_timeBuckets.length - 1] + _bucketSize.getSeconds()) {
-      return -1;
-    }
-    return (int) ((timeValue - _timeBuckets[0]) / _bucketSize.getSeconds());
+    long offsetFromRangeStart = timeValue - getTimeRangeStartExclusive();
+    return (int) ((offsetFromRangeStart - 1) / _bucketSize.getSeconds());
```

Comment:
nit: We should add some commentary for -1 math, just for new readers to understand.